### DISCUSSION TITLE
Using the shovel, m will not forward.

### DISCUSSION THREAD

**Discussion by springmvcGitHub** [rabbitmq/rabbitmq-server#7254] [Other]
hi rabbitmq expert,When I use the shovel to forward broker A's data from amq.topic+aa(routingKey) to broker B's amq.topic+aa(routingKey), when A and B are connected to each other, data m will be transferred to B as soon as it is sent to A. However, if the network of A and B is disconnected, after data m is sent to A, the network of A and B is connected, data m will not be forwarded to B. Is this phenomenon correct? If I want to forward data m to B after A and B are connected, how can I solve this problem?

**Comment by michaelklishin** [comment]
@springmvcGitHub we have asked you multiple times before to use Discussions for questions, please cooperate.

Shovels should reconnect and re-publish the messages they have enqueued. I’d that’s not the case, we need an executable way to reproduce, for example, using Toxiproxy. We do not guess in this community.
